### PR TITLE
WIP: Improve Pins Performance

### PR DESCRIPTION
**Pull request by atatmaja**
- Only request lat/lng columns in order to render pins. Request the rest of the information when clicking to reveal the modal.

### CODE REVIEW

File: src/httpClient.ts [hunk 1]
```diff
@@ -96,49 +88,26 @@ export default class httpClient {
...
+            return mapToAirtableRecord(item);
         });
-        // Remove timestamp from updated at string
 
         return filtered_records;
     }
 
+    async getOneAirtableRecord(source_id: string) {
+        const response = await this.httpGet(`/data_provider/record_details/${source_id}`);
+        console.log(response);
```
Comment (joanna-chen) [inline src/httpClient.ts:107]: remove console log

Comment (joanna-chen) [inline src/httpClient.ts:107]: remove console log

File: src/utils/stateUpdateUtils.tsx [hunk 2]
```diff
@@ -36,7 +37,7 @@ export const updateFilters = async (
   const [records,
     estimateGradePrevalences, reAggregatedRecords] = await
       Promise.all([
-        api.getAirtableRecords(updatedFilters, exploreIsOpen),
+        api.getAirtableRecords(updatedFilters, exploreIsOpen ? pins_columns : all_columns),
```
Comment (joanna-chen) [inline src/utils/stateUpdateUtils.tsx:40]: Should be `pins_modal_columns`

File: src/utils/stateUpdateUtils.tsx [hunk 3]
```diff
@@ -85,7 +86,7 @@ export const initializeData = async (dispatch: any, filters: Filters, exploreIsO
   const [records,
     reAggregatedRecords,
     estimateGradePrevalences] = await Promise
-      .all([api.getAirtableRecords(filters, exploreIsOpen),
+      .all([api.getAirtableRecords(filters, exploreIsOpen ? pins_columns : all_columns),
```
Comment (joanna-chen) [inline src/utils/stateUpdateUtils.tsx:89]: Same with this one, `pins_modal_columns`

File: src/utils/stateUpdateUtils.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import _ from 'lodash';
 import httpClient from '../httpClient'
 import { AggregationFactor, Filters, FilterType } from '../types'
+import { pins_columns, all_columns } from '../constants';
```
Comment (joanna-chen) [inline src/utils/stateUpdateUtils.tsx:4]: Should be importing `pins_modal_columns`

File: src/components/map/PinModal.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import React from "react";
```
Comment (joanna-chen) [inline src/components/map/PinModal.tsx:1]: nit: could you change all these double indents to just a single indent (2 spaces)?

**Review by joanna-chen — CHANGES_REQUESTED** [review]
I tested this branch locally, it seems to be really slow for me and it comes back with several errors. Do you get this on your end?

File: src/utils/stateUpdateUtils.tsx (hunk 3 above)
Comment (atatmaja) [reply src/utils/stateUpdateUtils.tsx:89]: On load I actually only want to be pulling in pins_columns. Only when we click on the modal do I want to be pulling in all the pins_modal_columns

File: src/components/map/PinModal.tsx (hunk 5 above)
Comment (atatmaja) [reply src/components/map/PinModal.tsx:1]: > I tested this branch locally, it seems to be really slow for me and it comes back with several errors. Do you get this on your end?

@joanna-chen this change actually depends on some backend work to clean up our get/record_details endpoint (gets a single record upon specifying a record source_id). that's why I've made this PR a draft for now.

**Comment by atatmaja** [comment]
> I tested this branch locally, it seems to be really slow for me and it comes back with several errors. Do you get this on your end?

@joanna-chen this change actually depends on some backend work to clean up our get/record_details endpoint (gets a single record upon specifying a record source_id). that's why I've made this PR a draft for now.

**Comment by atatmaja** [comment]
Taking down this PR because it will be better for @brett-stephen to address this via his mapping client changes

**Pull request closed without merging**